src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowRunTest.java

```diff
@@ -465,4 +465,12 @@ private void assertCulprits(WorkflowRun b, String... expectedIds) throws IOExcep
         }
     }
 
+    @Issue("JENKINS-31096")
+    @Test public void unicode() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        String message = "¡Čau → there!";
+        p.setDefinition(new CpsFlowDefinition("echo '" + message + "'", true));
+        r.assertLogContains(message, r.buildAndAssertSuccess(p));
```

Comment:
On Windows (ISO-8859-1), this fails with mojibake with the `charset = "UTF-8"` line commented out, or with the upstream PR not integrated.